pkg/config/imports.go

```diff
@@ -0,0 +1,406 @@
+
+	// Recursively process imports from the remote file
+	if Imports != nil && len(Imports) > 0 {
+		nestedPaths, err := processImports(importBasePath, Imports, tempDir, currentDepth+1, maxDepth)
+		if err != nil {
+			log.Debug("failed to process nested imports", "import", importPath, "err", err)
+			return nil, fmt.Errorf("failed to process nested imports")
+		}
+		resolvedPaths = append(resolvedPaths, nestedPaths...)
+	}
+
+	return resolvedPaths, nil
+}
+
+// Process local imports
+func processLocalImport(basePath string, importPath, tempDir string, currentDepth, maxDepth int) ([]ResolvedPaths, error) {
+	if importPath == "" {
+		return nil, fmt.Errorf("import_path required to process imports")
+	}
+	if !filepath.IsAbs(importPath) {
+		importPath = filepath.Join(basePath, importPath)
+	}
+	if !strings.HasPrefix(filepath.Clean(importPath), filepath.Clean(basePath)) {
+		log.Warn("Import path is outside of base directory",
+			"importPath", importPath,
+			"basePath", basePath,
+		)
+	}
+	paths, err := SearchAtmosConfig(importPath)
+	if err != nil {
+		log.Debug("failed to resolve local import path", "path", importPath, "err", err)
+		return nil, ErrResolveLocal
+	}
+
+	resolvedPaths := make([]ResolvedPaths, 0)
+	// Load the local configuration file to check for further imports
+	for _, path := range paths {
+		v := viper.New()
+		v.SetConfigFile(path)
+		v.SetConfigType("yaml")
+		err := v.ReadInConfig()
+		if err != nil {
+			log.Debug("failed to load local config", "path", path, "error", err)
+			continue
+		}
+		resolvedPaths = append(resolvedPaths, ResolvedPaths{
+			filePath:    path,
+			importPaths: importPath,
+			importType:  LOCAL,
+		})
+		Imports := v.GetStringSlice("import")
+		importBasePath := v.GetString("base_path")
+		if importBasePath == "" {
+			importBasePath = basePath
+		}
+
+		// Recursively process imports from the local file
+		if Imports != nil {
+			nestedPaths, err := processImports(importBasePath, Imports, tempDir, currentDepth+1, maxDepth)
+			if err != nil {
+				log.Debug("failed to process nested imports from", "path", path, "error", err)
+				continue
+			}
+			resolvedPaths = append(resolvedPaths, nestedPaths...)
+		}
+	}
+
+	return resolvedPaths, nil
+}
+
+// SearchAtmosConfig searches for a config file in path. The path is directory, file or a pattern.
+func SearchAtmosConfig(path string) ([]string, error) {
+	if stat, err := os.Stat(path); err == nil {
+		if !stat.IsDir() {
+			return []string{path}, nil
+		}
+	}
+	// Generate patterns based on whether path is a directory or a file/pattern
+	patterns, err := generatePatterns(path)
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate search patterns: %w", err)
+	}
+	// Find files matching the patterns
+	atmosFilePaths, err := findMatchingFiles(patterns)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find matching files: %w", err)
+	}
+	// Convert paths to absolute paths
+	atmosFilePathsABS, err := convertToAbsolutePaths(atmosFilePaths)
```

Comment:
ABS is not an acronym but an abbreviation

```suggestion
	atmosFilePathsAbs, err := convertToAbsolutePaths(atmosFilePaths)
```